src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -948,7 +949,14 @@ public void spawnCitizen(final CitizenData data)
             CitizenData citizenData = data;
             if (citizenData == null)
             {
-                topCitizenId++;
+                for (int i = 1; i <= MAX_CITIZENS_COUNTER; i++)
```

Comment:
Why only count up to 100? Why not use the colonies dynamic value of maxCitizens currently allowed?